core/src/main/java/nl/nn/adapterframework/webcontrol/api/TestPipeline.java

```diff
@@ -89,27 +88,19 @@ public Response postTestPipeLine(MultipartFormDataInput input) throws ApiExcepti
 		InputStream file = null;
 		IAdapter adapter = null;
 		
-		Map<String, List<InputPart>> inputDataMap = input.getFormDataMap();
 		try {
-			if(inputDataMap.get("message") != null)
-				message = inputDataMap.get("message").get(0).getBodyAsString();
-			if(inputDataMap.get("encoding") != null)
-				fileEncoding = inputDataMap.get("encoding").get(0).getBodyAsString();
-			if(inputDataMap.get("adapter") != null) {
-				String adapterName = inputDataMap.get("adapter").get(0).getBodyAsString();
+			if(input.getAttachmentObject("message", String.class) != null)
+				message = input.getAttachmentObject("message", String.class);
+			if(input.getAttachmentObject("encoding", String.class) != null)
+				fileEncoding = input.getAttachmentObject("encoding", String.class);
+			if(input.getAttachment("adapter").getDataHandler().getDataSource().getName() != null) {
+				String adapterName = IOUtils.toString(input.getAttachment("adapter").getDataHandler().getDataSource().getInputStream());
```

Comment:
adapter should be a simple string?

also:
resolveStringFromMap(inputDataMap, "message");

Since encoding is optional use resolveStringFromMap(inputDataMap, "encoding", Misc.DEFAULT_INPUT_STREAM_ENCODING);